Async demo and hacky fix

evolvedlight opened the pull request:
This addresses #298

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Excel-DNA/ExcelDna?pullRequest=299) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Excel-DNA/ExcelDna?pullRequest=299) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Excel-DNA/ExcelDna?pullRequest=299) it.</sub>

evolvedlight commented:
> Can you rather move the AsyncTrustedRTD add-in sample to the Excel-DNA\Samples project?

Sure, cleaned up and moved to https://github.com/Excel-DNA/Samples/pull/13

evolvedlight commented:
Hi @govert 
Did you have any thoughts on the end about how this can be actually fixed? The included was the best I had so far.
Regards,
Steve
